### PR TITLE
[CELEBORN-1477][CIP-9] Refine the celeborn RESTful APIs

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

This PR is for [CIP-9 Refine the celeborn RESTful APIs](https://docs.google.com/document/d/1LV2vV-w3XtlbJj2Vi4J77mt4IYCr40-8A_JncZLsHqs/edit?usp=sharing).

We leverage [openapi-generator](https://github.com/OpenAPITools/openapi-generator) to generate the client and model code.

### Why are the changes needed?

Celeborn has implemented RESTful APIs for monitoring and administrative operations on both master and worker endpoints. These APIs enable tasks such as configuration checks, status viewing of master/worker nodes, worker decommissioning/recommissioning, and more. They provide crucial insights and support for DevOps.
The primary concern with the existing API is the response content type, which is `text/plain` rather than the more widely accepted `application/json`. This mismatch makes integration with DevOps tools challenging, as these tools typically require JSON-formatted responses for seamless parsing and automation.
And I also saw the need for REST API evolution in[ Apache Celeborn CLI Proposal](https://cwiki.apache.org/confluence/display/CELEBORN/CIP-7+Celeborn+CLI).

### Does this PR introduce _any_ user-facing change?
This pr introduce  a new API namespace: `/api/v1`. This approach allows us to maintain the current API for compatibility while offering an improved version.

### How was this patch tested?
UT.

### CODE REVIEW

**Comment by turboFei** [comment]
cc @pan3793 @SteNicholas @RexXiong

**Comment by FMX** [comment]
@turboFei Hi, this PR has too many codes to review. Is there a solution to remove the auto-generated codes?

**Comment by turboFei** [comment]
> @turboFei Hi, this PR has too many codes to review. Is there a solution to remove the auto-generated codes?

Checking.

**Comment by turboFei** [comment]
> Is there a solution to remove the auto-generated codes?


To prevent copy the auto-generated codes, we have to use [sbt-openapi-generator](https://github.com/OpenAPITools/sbt-openapi-generator) 5.0.1, which is the only version supports JDK8 and outdated.

To reduce the efforts to maintain both maven and sbt plugin, I spilt the openapi model into two modules:
- openapi-client: independent and not depended by any module, and only supports to build with maven plugin
- openapi-model: depended by celeborn-service module and supports to build both with maven and sbt plugin

**Comment by turboFei** [comment]
Hi @FMX I have removed the generated code.

**Comment by turboFei** [comment]
I setup the openapi-client module for sbt as well.

There is no gap between maven and sbt now.

**Comment by turboFei** [comment]
also cc @akpatnam25, I saw you are working on the CIP 7: Celeborn CLI.

File: pom.xml [hunk 1]
```diff
@@ -116,6 +118,9 @@
     <jersey.version>2.39.1</jersey.version>
     <jetty.version>9.4.52.v20230823</jetty.version>
     <jakarta.servlet-api.version>4.0.4</jakarta.servlet-api.version>
+    <!-- openapi-generator dependencies -->
+    <openapitools.jackson-databind-nullable.version>0.2.6</openapitools.jackson-databind-nullable.version>
+    <openapitools.swagger.version>1.6.11</openapitools.swagger.version>
```
Comment (turboFei) [inline pom.xml:123]: For maven, the openapi-generator version is 6.6.0, the latest version supports Java 8.
For sbt, the sbt-openapi-generator version is 5.0.1, the only version supports Java 8.

And due to openapi-generator 5.0.1 does not support <annotationLibrary>none</annotationLibrary> and <annotationLibrary>swagger2</annotationLibrary>,, so we have to use introduce swagger1 dependencies.
FYI:
https://openapi-generator.tech/docs/generators/java

We can remove swagger1.x dependencies after dropping the support for JDK8 and upgrade the openapi-generator to 7.x+ in the future.

**Comment by turboFei** [comment]
cc @FMX @pan3793 , I have self reviewed for several times, it should be fine now.

File: pom.xml (hunk 1 above)
Comment (pan3793) [reply pom.xml:123]: thanks for investigating, I agree the descision

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Overall LGTM, the API audit can be done during the CLI developing phase

**Comment by pan3793** [comment]
Thanks, merged to main for 0.6

**Pull request closed without merging**